text/metrics/0156-counter-gauge-model.md

```diff
@@ -0,0 +1,356 @@
+aggregation function to merge metric streams.
+
+For example, any metric event with no attributes:
+
+```
+gauge.Set(value)
+```
+
+can be extended by a new attribute, without changing its meaning or
+altering any existing interpretation:
+
+```
+gauge.Set(value, { 'property': this.property })
+```
+
+## New measurements: Counter and UpDownCounter instruments
+
+Sum points have been defined to have linear scale of measurement,
+therefore Sum points can be subdivided.  A single Counter event can be
+logically replaced by multiple Counter events having an equal sum.
+This property allows the producer of metric events to introduce new
+measurements, while preserving existing interpretation.
+
+For example, it is reasonable to replace a single Counter event adding `x+y`:
+
+```
+counter.Add(x+y)
+```
+
+with separate counter events and one additional attribute:
+
+```
+counter.Add(x, { 'property': 'X' })
+counter.Add(y, { 'property': 'Y' })
+```
+
+This property for Sum points makes it possible to configure an
+instrumentation library with or without subdivided Sums and to
+meaningfully aggregate data with a mixture of attributes.
+
+## New measurements: Gauge instruments
+
+Gauge instruments, unlike Counter instruments, cannot be subdivided.
+Multiple Gauge measurements cannot be meaningfully combined using
+addition.  In the time dimension, Gauge instrument events are
+aggregated by taking the last value.
+
+The same aggregation can be applied when removing an attributes from
+metric streams forces reaggregation.  The most current value should be
+selected.  In case of identical timestamps, a random value should be
+selected to preserve the meaning of the Gauge.
+
+For example, a Gauge for expressing a vehicle's speed relative to the
+ground can be expressed either as the speed of its midpoint or by an
+independent measurement of the speed of each wheel.
+
+```
+speedGauge.Set(vehicleSpeed)
+```
+
+This can be replaced by one Gauge per wheel, since wheel speed and
+vehicle speed each define vehicle speed relative to the ground:
+
+```
+for i := 0; i < 4; i++ {
+  speedGauge.Set(wheelSpeed[i], { 'wheel': i })
+}
+```
+
+This form of Gauge rewrite is generally useful to capture additional
+measurements by creating distinct metric streams.
+
+## Meaning-preserving attribute erasure
+
+Several rules for rewriting metric events that preserve meaning have
+been shown above, focused on introducing new attributes and new
+measurements in ways do not change existing meaning or alter existing
+interpretations.
+
+Removing attributes from metric events does not, by definition, change
+their meaning, since attributes are interpreted as event selectors.
+Removing attributes from aggregated streams of OpenTelemetry Metrics
+data requires attention to the meaning being conveyed.
+
+Safe attribute erasure for OpenTelemetry Metrics streams is specified
+in a way that preserves meaning while removing only the forms of
```

Comment:
What if "removing the forms of interpretation that made use of the erased attribute" basically is destroying the practical value of the metric?


I think this is implying that removing an attribute from a metric is an ok thing to do (it' s not).  The meaning that IS preserved is relevant to reaggregation + processing, but could be disastrous to dashboards and users.